### PR TITLE
Select Runner Label Dynamically

### PR DESCRIPTION
**Pull request by DanilBaibak**
Updated `get_workflow_type.py` logic to dynamically select a prefix for the runner label.

### CODE REVIEW

**Review by ZainRizvi — CHANGES_REQUESTED** [review]
This schema would be pretty good if we wanted to have the LF runners always use the same naming convention as the existing runners but with a `lf.` prefix attached.  While that schema is okay for temporary testing, as discussed in Slack it's not something we really want to stick with long term

Instead, it would be great to have some sort of a mapping file, that maps from the old runner name to the new runner name, allowing us to chose any arbitrary text for the new runner names. 

One way it could work:

This action returns a dictionary that maps original label name -> desired label name
E.g. for people opted into the migration, it'll map "linux.2xlarge" -> "lf.linux.2xlarge" but to everyone else it'll map "linux.2xlarge" -> "linux.2xlarge".  

Then the jobs will look in that dictionary for the value corresponding to their original label.

File: .github/scripts/get_workflow_type.py [hunk 1]
```diff
@@ -49,47 +48,34 @@ def is_exception_branch(branch: str) -> bool:
...
 
 def main() -> None:
     args = parse_args()
 
     if is_exception_branch(args.github_branch):
-        output = {"workflow_type": WORKFLOW_TYPE_LABEL}
+        output = {"label_type": WORKFLOW_LABEL_META}
     else:
         try:
             gh = get_gh_client(args.github_token)
```
Comment (ZainRizvi) [inline .github/scripts/get_workflow_type.py:73]: nit: can you add a comment linking to the default issue that is used here? It'll save people time searching for it and we're unlikely to need to use another issue anyways except possibly for testing

File: .github/scripts/get_workflow_type.py [hunk 2]
```diff
@@ -49,47 +48,34 @@ def is_exception_branch(branch: str) -> bool:
...
             issue = get_issue(gh, args.github_repo, args.github_issue)
 
-            output = {"workflow_type": get_workflow_type(issue, args.github_user)}
+            output = {"label_type": get_workflow_type(issue, args.github_user)}
```
Comment (ZainRizvi) [inline .github/scripts/get_workflow_type.py:76]: nit: it's not related to this PR but can you add a comment linking to the default issue that is used here? It'll save people time searching for it and we're unlikely to need to use another issue anyways except possibly for testing

**Comment by DanilBaibak** [comment]
I think the purpose of dynamic rollout is slightly different at the current point. We want to be able quickly switch between different sources of runners (meta and linux foundation) only at the stage of the testing. Once we can run a full load on the LF account, we will need to do a cleanup and remove the [dynamic labelling](https://github.com/pytorch/pytorch/pull/127287/files#diff-375be920291dc9bd5736cae7dc66c3896ade96f1abae1096286520c772135ca4R50).

**Comment by DanilBaibak** [comment]
Tested previously:
Screenshot 2024-06-11 at 10 23 05

File: .github/scripts/get_workflow_type.py [hunk 3]
```diff
@@ -49,47 +48,35 @@ def is_exception_branch(branch: str) -> bool:
...
-            # Use ARC runners and old runners for everyone
-            return WORKFLOW_TYPE_BOTH
-        else:
-            # Use only ARC runners for everyone
-            return WORKFLOW_TYPE_RG
-    elif username in user_list:
-        if run_option == WORKFLOW_TYPE_BOTH:
-            # Use ARC runners and old runners for a specific user
-            return WORKFLOW_TYPE_BOTH
+        user_list = issue.get_comments()[0].body.split("\r\n")
```
Comment (ZainRizvi) [inline .github/scripts/get_workflow_type.py:52]: I'm surprised to see `\r` included in the splitter.

Seems like a subtle bug ready to fire if the wrong tool/OS updates the comment

What do you think about spliting it on just `\n`, and then trimming each resulting entry for `\r`?

Comment (ZainRizvi) [inline .github/scripts/get_workflow_type.py:52]: also, is the indexing correct here?

```suggestion
        user_list = issue.get_comments()[1].body.split("\r\n")
```

File: .github/scripts/get_workflow_type.py [hunk 4]
```diff
@@ -49,47 +48,35 @@ def is_exception_branch(branch: str) -> bool:
...
+    except Exception as e:
+        return WORKFLOW_LABEL_META
 
 
 def main() -> None:
     args = parse_args()
 
     if is_exception_branch(args.github_branch):
-        output = {"workflow_type": WORKFLOW_TYPE_LABEL}
+        output = {"label_type": WORKFLOW_LABEL_META}
```
Comment (ZainRizvi) [inline .github/scripts/get_workflow_type.py:70]: nit: how about turning `"label_type"` into a const?

**Review by ZainRizvi — APPROVED** [review]
For phase 0, we should shrink down the list of people opted-in to just us ALI folks before enabling this for any PR, to limit the blast radius in case it goes horribly wrong

File: .github/scripts/get_workflow_type.py (hunk 3 above)
Comment (DanilBaibak) [reply .github/scripts/get_workflow_type.py:52]: No, the first block is "body".  After the body we have a list of comments.

Comment (DanilBaibak) [reply .github/scripts/get_workflow_type.py]: I double-checked, we don't even need "\r\n".

**Pull request closed without merging**